Internal/KeyboardObserver.swift

```diff
@@ -0,0 +1,168 @@
+//
+//  KeyboardObserver.swift
+//  BlueprintUICommonControls
+//
+//  Created by Kyle Van Essen on 2/16/20.
+//
+
+import UIKit
+
+
+protocol KeyboardObserverDelegate : AnyObject
+{
+    func keyboardFrameWillChange(observer : KeyboardObserver)
+}
+
+/**
+ Implementation borrowed from Listable:
+ https://github.com/kyleve/Listable/blob/master/Listable/Sources/Internal/KeyboardObserver.swift
+ 
+ iOS Docs for keyboard management:
+ https://developer.apple.com/library/archive/documentation/StringsTextFonts/Conceptual/TextAndWebiPhoneOS/KeyboardManagement/KeyboardManagement.html
+ */
+final class KeyboardObserver
+{
+    private let center : NotificationCenter
+    
+    weak var delegate : KeyboardObserverDelegate?
+    
+    //
+    // MARK: Initialization
+    //
+    
+    init(center : NotificationCenter = .default)
+    {
+        self.center = center
+
+        self.center.addObserver(self, selector: #selector(keyboardFrameChanged(_:)), name: UIWindow.keyboardWillChangeFrameNotification, object: nil)
+        self.center.addObserver(self, selector: #selector(keyboardFrameChanged(_:)), name: UIWindow.keyboardDidChangeFrameNotification, object: nil)
+    }
+    
+    private var latestNotification : NotificationInfo?
+    
+    //
+    // MARK: Handling Changes
+    //
+    
+    enum KeyboardFrame : Equatable
+    {
+        case notVisible
```

Comment:
nit: would name this noOverlap, or notInView. notVisible makes it sound like its not on screen at all, when it might be, but just not overlapping the passed in view.

Reply:
I went back and forth on this, but found the other things awkward... Let me see if I can come up with something better.